primitives/sidechain-mc-hash/src/lib.rs

```diff
@@ -201,6 +207,38 @@ impl InherentDigest for McHashInherentDigest {
 	}
 }
 
+pub fn inherent_digest_value_for_hash<ID: InherentDigest, Block: BlockT, C>(
```

Comment:
This is something that should ideally reside with the `InherentDigest` trait itself, but I added this here for now

Reply:
1) Regarding your comment: I'm not sure if this is generic enough. Why for block number zero there is None returned? Maybe some implementations could get it out from genesis?

2) method name suggests it can get digest from hash, I suggest `inherent_digest_from_block`.

Reply:
Genesis block has an empty digest log and there's no mechanism that can put an inherent digest there.